packages/react-sdk/src/ReactSdkThemeProvider.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2025 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { MuiThemeProvider } from '@matrix-widget-toolkit/mui';
+import { CssBaseline, ThemeProvider, useTheme } from '@mui/material';
+import { deepmerge } from '@mui/utils';
+import { PropsWithChildren, useMemo } from 'react';
+import theme from './theme';
+
+/**
+ * This component wraps the children with both MuiThemeProvider and ThemeProvider,
+ * applying the custom fonts and theme settings from the `theme.ts` file.
+ */
+export function ReactSdkThemeProvider({ children }: PropsWithChildren<{}>) {
+  const MuiTheme = useTheme();
+
+  const reactSdkTheme = useMemo(() => {
```

Comment:
This should be a function (/callback). See extending themes here https://mui.com/system/styles/advanced/#theme-nesting . In the code we only need one theme provider then.

Also, can you please check if we can remove the additional theme provider in the widget?

Reply:
I will change it to useCalback. I used useMemo as example from standalone [here](https://github.com/nordeck/matrix-neoboard-standalone/blob/203dd405b88816ed27571df091d4597a400731af/src/components/StandaloneThemeProvider/StandaloneThemeProvider.tsx#L47). 
In this code we have just one themeProvider in `ReactSdkThemeProvider` as nested from toolkit so i dont know what do u mean with `Also, can you please check if we can remove the additional theme provider in the widget?`8

Reply:
For PDF there is separate ticket

Reply:
> For PDF there is separate ticket

Unfortunately I can't follow what PDF has to in the context of this discussion.

Reply:
> Also, can you please check if we can remove the additional theme provider in the widget?

This is also net yet done. We have two providers now. Let's have a look at this together.